tests/test_main.py

```diff
@@ -1,6 +1,136 @@
-# TODO: add tests
-from app.main import hello_world
+import pytest
 
+from app.main import make_a_person
+from app.main import Person
 
-def test_hello_world():
-    assert hello_world() == "Hello, world!"
+
+def test_person_class_attribute_people_exists():
+    assert hasattr(Person, 'people'), (
+        "Class Person should have class attribute 'people'"
+    )
+    assert len(Person.people) == 0, (
+        "Initial length of 'Person.people' should equal to 0"
+    )
+
+
+@pytest.mark.parametrize(
+    'person,name,age',
+    [
+        ({'name': 'Ross', 'age': 30, 'wife': 'Rachel'}, 'Ross', 30),
+        ({'name': 'Joey', 'age': 29, 'wife': None}, 'Joey', 29)
+    ]
+)
+def test_person_class_name_age(person, name, age):
+    person_inst = Person(person['name'], person['age'])
+    assert person_inst.name == name, (
+        f"Attribute 'name' for Person instance for {person} "
+        f"should equal to {name}"
+    )
+    assert person_inst.age == age, (
+        f"Attribute 'age' for Person instance for {person} "
+        f"should equal to {age}"
+    )
+
+
+def test_make_a_person_all_persons():
+    people = [
+        {'name': 'Ross', 'age': 30, 'wife': 'Rachel'},
+        {'name': 'Joey', 'age': 29, 'wife': None},
+        {'name': 'Phoebe', 'age': 31, 'husband': None},
+        {'name': 'Chandler', 'age': 30, 'wife': 'Monica'},
+        {'name': 'Monica', 'age': 32, 'husband': 'Chandler'},
+        {'name': 'Rachel', 'age': 28, 'husband': 'Ross'},
+    ]
+    person_list = make_a_person(people)
+    assert all(isinstance(person, Person) for person in person_list), (
+        "All elements in result of 'make_a_person' should be instance "
+        "of Person class"
+    )
+    assert len(person_list) == len(people), (
+        "Length of initial list should equal to length of function result"
+    )
+
+
```

Comment:
A lot of code duplication like these (#36 - #44 lines of code) - here should be a fixture:
```python
@pytest.fixture()
def people_data():
    return [
        {"name": "Ross", "age": 30, "wife": "Rachel"},
        {"name": "Joey", "age": 29, "wife": None},
        {"name": "Phoebe", "age": 31, "husband": None},
        {"name": "Chandler", "age": 30, "wife": "Monica"},
        {"name": "Monica", "age": 32, "husband": "Chandler"},
        {"name": "Rachel", "age": 28, "husband": "Ross"},
    ]


@pytest.fixture()
def created_person_list(people_data):
    return create_person_list(people_data)


def test_create_person_list_order(people_data, created_person_list):
    assert [person_dict["name"] for person_dict in people_data] == [
        person.name for person in created_person_list
    ], "Order in function result should be the same"
```